### ISSUE TITLE
Many features result in GUI stutter on call of `get_value`

### ISSUE THREAD

**Issue by brisvag** [napari/napari#5177]
## 🐛 Bug

Having many features on a points layer results in gui stutter (or worse) every time the mouse hovers a point, likely because of some issue with `get_value`.

## To Reproduce

This is buggy:
```
import napari
import numpy as np
import pandas as pd

df = pd.DataFrame.from_records(np.random.rand(100, 100))

v = napari.Viewer()
v.add_points(np.random.rand(100, 2) * 100, features=df)
```

now move your mouse over the points. In my tiling window manager, this is what I see if the window is tiled:

https://user-images.githubusercontent.com/23482191/194061552-1df16e93-33e9-4ded-86d8-cdc1f3570d0e.mp4

And even worse, when the window is floating it just explodes to beyond the screen limits:

https://user-images.githubusercontent.com/23482191/194061628-76ec29cc-335c-40b2-abf6-72afb0d0dfd5.mp4

## Possible solution

Simply ensure the size of the returned message is smaller than the window is probably enough?

**Comment by aganders3** [comment]
I was able to reproduce this (thanks for your script) and _even worse_ it simply crashed for me.

I hacked together a quick POC to confirm the fix you proposed should work. This just tweaks how the coordinates are set in the status bar so the text is elided.

https://user-images.githubusercontent.com/1231828/194197472-609158f6-90ac-470b-a5de-e871ca33453c.mov

Unfortunately for this I had to manage the eliding myself because the `superqt.QElidingLabel` had some other weird behavior where it could slowly grow when moving the mouse quickly from point-to-point. This might just be a bug in superqt (or I'm using it wrong) so I'll probably keep testing it.

**Comment by brisvag** [comment]
Thanks for the input @aganders3! Maybe our resident qt wizard @Czaki has some suggestions about `QElidingLabel`? I know we had a *very* lengthy discussions about the satus bar in #4664, so maybe it's worth pinging that as well.

**Comment by Czaki** [comment]
> Thanks for the input @aganders3! Maybe our resident qt wizard @Czaki has some suggestions about `QElidingLabel`? I know we had a _very_ lengthy discussions about the satus bar in #4664, so maybe it's worth pinging that as well.

From my point of view, QElidingLabel is bugged and may require to reimplementing from scratch. 
For example, when you set the text, it does not reduce the width of elited label. And this may be connected with small increase of size when moving mouse. 

But I do not see why it may cause drastically increase in size. I was sure that this was fixed in #4980.

**Comment by aganders3** [comment]
> For example, when you set the text, it does not reduce the width of elited label. And this may be connected with small increase of size when moving mouse.

This is consistent with what I observed. There was no drastic increase in size with `QElidingLabel`. But there are still a few labels in the status bar that use `QLabel` and the reproducer above demonstrates at least `_coordinates` might get a rather long input.

```python
# from `qt_viewer_status_bar.py
class ViewerStatusBar(QStatusBar):
    def __init__(self, parent: '_QtMainWindow') -> None:
        ...
        self._status = QLabel(trans._('Ready'))
        self._layer_base = QElidingLabel(trans._(''))
        self._plugin_reader = QElidingLabel(trans._(''))
        self._source_type = QLabel('')
        self._coordinates = QLabel('')
        self._help = QElidingLabel('')
        ...
```

Another solution I've seen that would prevent the root behavior (window size change on status bar size change) is to put the whole status bar into a `QScrollArea` with hidden scrollbars.

**Comment by brisvag** [comment]
Closed by #5451.
